pkg/trace/api/profiles.go

```diff
@@ -22,48 +25,78 @@ const (
 	profilingURLDefault = "https://intake.profile.datadoghq.com/v1/input"
 )
 
-// profilingEndpoint returns the profiling intake API URL based on agent configuration.
-func profilingEndpoint() string {
+// profilingEndpoints returns the profiling intake urls and their corresponding
+// api keys based on agent configuration. The main endpoint is always returned as
+// the first element in the slice. When multiple endpoints are in use, the
+// response from the main endpoint is proxied back to the client, while for
+// all aditional endpoints the response is discarded. There is no de-duplication
+// done between endpoint hosts or api keys.
+func profilingEndpoints(apiKey string) []*traceconfig.Endpoint {
+	endpoints := []*traceconfig.Endpoint{}
+	var main string
 	if v := config.Datadog.GetString("apm_config.profiling_dd_url"); v != "" {
-		return v
+		main = v
+	} else if site := config.Datadog.GetString("site"); site != "" {
+		main = fmt.Sprintf(profilingURLTemplate, site)
+	} else {
+		main = profilingURLDefault
 	}
-	if site := config.Datadog.GetString("site"); site != "" {
-		return fmt.Sprintf(profilingURLTemplate, site)
+	endpoints = append(endpoints, &traceconfig.Endpoint{Host: main, APIKey: apiKey, NoProxy: false})
+
+	if config.Datadog.IsSet("apm_config.profiling_additional_endpoints") {
+		extra := config.Datadog.GetStringMapStringSlice("apm_config.profiling_additional_endpoints")
+		for endpoint, keys := range extra {
+			for _, key := range keys {
+				endpoints = append(endpoints, &traceconfig.Endpoint{Host: endpoint, APIKey: key, NoProxy: false})
+			}
+		}
 	}
-	return profilingURLDefault
+
+	return endpoints
 }
 
-// profileProxyHandler returns a new HTTP handler which will proxy requests to the profiling intake.
-// If the URL can not be computed because of a malformed 'site' config, the returned handler will always
+// profileProxyHandler returns a new HTTP handler which will proxy requests to the profiling intakes.
+// If the main intake URL can not be computed because of config, the returned handler will always
 // return http.StatusInternalServerError along with a clarification.
 func (r *HTTPReceiver) profileProxyHandler() http.Handler {
-	target := profilingEndpoint()
-	u, err := url.Parse(target)
-	if err != nil {
-		log.Errorf("Profile forwarder is OFF because of invalid intake URL: %v", err)
-		return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-			msg := fmt.Sprintf("Agent is misconfigured with an invalid intake URL: %q", target)
-			http.Error(w, msg, http.StatusInternalServerError)
-		})
+	endpoints := profilingEndpoints(r.conf.APIKey())
+	targets := []*url.URL{}
+	keys := []string{}
```

Comment:
```suggestion
var (
    targets []*url.URL
    keys    []string
)
```
No need to initialize, `append` will do it.